### PR TITLE
Added some ports

### PR DESCRIPTION
**Pull request by Dennisbonke**
This PR adds the following ports

- `gmp`
- `mpfr`
- `mpc`

And fixes the following port
- `gcc`

### CODE REVIEW

File: bootstrap.yml [hunk 1]
```diff
@@ -40,6 +40,19 @@ sources:
...
+    git: 'git://gcc.gnu.org/git/gcc.git'
+    tag: 'releases/gcc-9.2.0'
+    tools_required:
+      - host-autoconf-v2.69
+      - host-automake-v1.11
+    regenerate:
+      - args: ['autoconf']
+        workdir: '@THIS_SOURCE_DIR@/gcc'
+      - args: ['autoconf']
+        workdir: '@THIS_SOURCE_DIR@/libstdc++-v3'
```
Comment (avdgrinten) [inline bootstrap.yml:54]: Why do we need a second source for GCC?

File: patches/gcc/0002-Managarm-target.patch [hunk 2]
```diff
@@ -87,6 +87,10 @@ index ddd3b8f4d..9701267b9 100644
 +    tmake_file="${tmake_file} t-slibgcc"
 +    thread_file='posix'
 +    ;;
++  *-managarm-pc*)
++    tmake_file="${tmake_file} t-slibgcc"
++    thread_file='posix'
++    ;;
```
Comment (avdgrinten) [inline patches/gcc/0002-Managarm-target.patch:93]: Why do we need `managarm-pc`?

Comment (avdgrinten) [reply patches/gcc/0002-Managarm-target.patch:93]: (Not saying that it's wrong, I'm just surprised that another suffix is needed.)

Comment (Dennisbonke) [reply patches/gcc/0002-Managarm-target.patch:93]: Gcc defaults to this suffix with the current configuration options. We can probably get it to accept managarm-system there as well but something-pc is pretty universal as a suffix

Comment (avdgrinten) [reply patches/gcc/0002-Managarm-target.patch:93]: OK.

**Comment by avdgrinten** [comment]
Thank you for the PR!

**Pull request merged**